src/main/scala/com/gu/contentapi/firehose/kinesis/KinesisStreamReader.scala

```diff
@@ -0,0 +1,53 @@
+package com.gu.contentapi.firehose.kinesis
+
+import java.util.concurrent.Executors
+
+import com.amazonaws.auth.AWSCredentialsProvider
+import com.amazonaws.services.kinesis.metrics.impl.NullMetricsFactory
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import java.util.UUID
+import com.amazonaws.services.kinesis.clientlibrary.lib.worker.{ Worker, KinesisClientLibConfiguration, InitialPositionInStream }
+import com.amazonaws.services.kinesis.clientlibrary.interfaces.IRecordProcessorFactory
+
+trait KinesisStreamReader {
+
+  val streamName: String
```

Comment:
There are so many pieces of configuration here, I'd be tempted to gather them into a `KinesisStreamReaderConfig` case class. The client code would pass an instance of the case class into the constructor of `ContentApiFirehoseConsumer`.